### ISSUE TITLE
ListView doesn't get updated when items in a group are added or removed on UWP 

### ISSUE THREAD

**Issue by rheesbeen** [xamarin/Xamarin.Forms#14174]
### Description
When you have a ListView with grouping enabled and the items in a group change (added or removed) the ListView doesn't update the items properly. They will still be visible if removed or not visible when added.

### Steps to Reproduce

1. Create a grouped ListView
2. Change items in the group source.
3. Notice they will not appear or be removed.

### Expected Behavior
I expect that items added or removed would work (this works for iOS and Android).

### Actual Behavior
Added or Removed items aren't being properly updated. Do not become visible or aren't removed.

### Basic Information

- Version with issue: Version 5.0.0.
- Last known good version: None
- Platform Target Frameworks: UWP

### Environment
Visual Studio 16.9.3

### Workaround
None

**Comment by rheesbeen** [comment]
What are my changes that this will be fixed? I tried pulling in the renderer but there are a few calls to internal properties/methods.

**Comment by ValonK** [comment]
Have you tried switching to `CollectionView` ? The chances that this will get fixed are low I think

**Comment by rheesbeen** [comment]
Yes, it has the same exact issue. I managed to pull in the renderer, but then nothing gets rendered at all. It's not calling my custom templateselector. And it says it can't find the property CachingStrategy.

It's really frustrating as it is works fine on Android and iOS. Unfortunately I'm getting used to it by now.

**Comment by rheesbeen** [comment]
Does anybody know if this would be fixed in MAUI? Or are all current bugs going to be present there as well?

**Comment by jsuarezruiz** [comment]
@rheesbeen Does it happen with any Xamarin.Forms 5.0 version?, Could you attach a small example where reproduce the issue?

**Comment by rheesbeen** [comment]
@jsuarezruiz Thnx for reaching out. Yes it happens on all the versions. I could try to make a sample if needed and you are planning on fixing the bug? I can also send you a movie which shows you what happens. I think that might help, but can I send it to you privately?

**Comment by Redth** [comment]
Please attach a repro sample if you're able to.
